sdk/scheduler/src/main/java/org/apache/mesos/specification/DefaultServiceSpecification.java

```diff
@@ -98,9 +131,13 @@ public String toJsonString() throws Exception {
     /**
      * Returns a {@link ConfigurationFactory} which may be used to deserialize
      * {@link DefaultServiceSpecification}s.
+     *
+     * @param additionalSubtypesToRegister any class subtypes which should be registered with
+     *     Jackson for deserialization. any custom placement rule implementations must be provided
      */
-    public static ConfigurationFactory<ServiceSpecification> getFactoryInstance() {
-        return FACTORY;
+    public static ConfigurationFactory<ServiceSpecification> getFactory(
```

Comment:
It would be very cool, if a unit test could be added validating that an attempt to use an unregistered (but properly constructed) custom PlacementRule in a ServiceSpecification caused some kind of validation error.  Otherwise I think people will run into faults at deserialization time i.e. after an unplanned Task crash or something and find out that things are never recoverable.